v2/tests/test_runner.py

```diff
@@ -1,42 +1,104 @@
         robot_target=pathlib.Path("~/suite/calculator.robot"),
         outputdir=pathlib.Path("/tmp/outputdir/"),
         output_name="1",
         previous_output=pathlib.Path("/tmp/outputdir/0.xml"),
         variablefile=None,
         argumentfile=pathlib.Path("~/suite/retry_arguments"),
-        retry_strategy=runner._RetryStrategy.INCREMENTAL,  # pylint: disable=protected-access
+        retry_strategy=runner._RetryStrategy.INCREMENTAL,
     )
-    command = runner._create_command(spec)  # pylint: disable=protected-access
+    command = runner._create_variant_command(spec)
     expected = (
         "/usr/bin/python3 -m robot --argumentfile=~/suite/retry_arguments "
         "--rerunfailed=/tmp/outputdir/0.xml --outputdir=/tmp/outputdir "
         "--output=/tmp/outputdir/1.xml ~/suite/calculator.robot"
     )
     assert command == expected
+
+
+def test_create_merge_command() -> None:
+    assert runner._create_merge_command(
+        python_executable=pathlib.Path("/usr/bin/python3"),
+        variants_outputs=[
+            pathlib.Path("/tmp/outputdir/0.xml"),
+            pathlib.Path("/tmp/outputdir/1.xml"),
+        ],
+        final_output=pathlib.Path("/tmp/outputdir/merged.xml"),
+    ) == (
+        "/usr/bin/python3 -m robot.rebot --output=/tmp/outputdir/merged.xml --report=NONE "
+        "--log=NONE /tmp/outputdir/0.xml /tmp/outputdir/1.xml"
+    )
+
+
+def test_create_commands() -> None:
+    assert runner._create_commands(
+        runner._RetrySpec(
+            id_=uuid.UUID("383783f4-1d02-43b1-9d6f-205f4d492d95"),
+            python_executable=pathlib.Path("/usr/bin/python3"),
+            robot_target=pathlib.Path("~/suite/calculator.robot"),
+            working_directory=pathlib.Path("/tmp/outputdir/"),
+            schedule=[
+                runner._Variant(
+                    variablefile=None,
+                    argumentfile=None,
+                ),
+                runner._Variant(
+                    variablefile=pathlib.Path("~/suite/retry.yaml"),
+                    argumentfile=None,
+                ),
+            ],
+            strategy=runner._RetryStrategy.INCREMENTAL,
+        )
+    ) == runner._RetryCommands(
+        variants=[
+            (
+                "/usr/bin/python3 -m robot "
+                "--outputdir=/tmp/outputdir/383783f41d0243b19d6f205f4d492d95 "
+                "--output=/tmp/outputdir/383783f41d0243b19d6f205f4d492d95/0.xml "
+                "~/suite/calculator.robot"
+            ),
+            (
+                "/usr/bin/python3 -m robot --variablefile=~/suite/retry.yaml "
+                "--rerunfailed=/tmp/outputdir/383783f41d0243b19d6f205f4d492d95/0.xml "
+                "--outputdir=/tmp/outputdir/383783f41d0243b19d6f205f4d492d95 "
+                "--output=/tmp/outputdir/383783f41d0243b19d6f205f4d492d95/1.xml "
+                "~/suite/calculator.robot"
+            ),
+        ],
+        merge=(
+            "/usr/bin/python3 -m robot.rebot "
```

Comment:
Happen happens if we have RetryStrategy, which runs twice, but has three variants specified? I believe the interface of the merge command is the result of command executions, rather than of the strategy?